testnet/launcher/docker.go

```diff
@@ -206,40 +207,42 @@ func waitForHealthyNode(port int) error { // todo: hook the cfg
 	reqBody := `{"method": "obscuro_health", "id": 1}`
 
 	fmt.Println("Waiting for obscuro node to be healthy...")
-	for startTime := time.Now(); time.Now().Before(startTime.Add(2 * time.Minute)); time.Sleep(time.Second) {
-		req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, requestURL, bytes.NewBufferString(reqBody))
-		if err != nil {
-			return fmt.Errorf("client: could not create request: %w", err)
-		}
-		req.Header.Set("Content-Type", "application/json")
-
-		res, err := http.DefaultClient.Do(req)
-		if err != nil {
-			continue
-		}
-		defer res.Body.Close()
-
-		resBody, err := io.ReadAll(res.Body)
-		if err != nil {
-			continue
-		}
-
-		response := map[string]interface{}{}
-		err = json.Unmarshal(resBody, &response)
-		if err != nil {
-			continue
-		}
-
-		if r := response["result"]; r != nil { //nolint: nestif
-			if h, ok := r.(map[string]interface{}); ok {
-				if overallHealth := h["OverallHealth"]; overallHealth != nil {
-					if health, ok := overallHealth.(bool); ok && health {
-						fmt.Println("obscuro node is ready")
-						return nil
+	return retry.Do(
+		func() error {
+			req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, requestURL, bytes.NewBufferString(reqBody))
+			if err != nil {
+				return fmt.Errorf("client: could not create request: %w", err)
```

Comment:
Maybe want to wrap this one in the `retry.FailFast(err)` wrapper, so it doesn't keep retrying to keep the behaviour you had before

Reply:
I might be getting this wrong. Please let me know.
I want it to never fail until it hits the sucess / timeout. Because the node might be unresponsive or might not be healthy yet it will throw errors, that we will ignore.

Does that make sense ?

Reply:
Yes that makes sense.

The way you had it before it would do that, except for errors from the NewRequestWithContext call, in that case you returned the error instead of doing a `continue` to try again.

I was just saying if you wanted to keep that behaviour for that one error check, then you could wrap it in a FailFast. But maybe you've decided that wasn't the right behaviour and better to always retry.

Reply:
Ahh I see. It's a moot. The `http.NewRequestWithContext` should never fail.